src/server_manager/web_app/gcp_account.ts

```diff
@@ -12,16 +12,103 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as gcp_api from '../cloud/gcp_api';
+import {SCRIPT} from '../install_scripts/gcp_install_script';
 import * as gcp from '../model/gcp';
+import * as server from '../model/server';
 
+import {GcpServer} from './gcp_server';
+
+/**
+ * The Google Cloud Platform account model.
+ */
 export class GcpAccount implements gcp.Account {
-  constructor(private refreshToken: string) {}
+  private static readonly OUTLINE_FIREWALL_NAME = 'outline';
+  private static readonly MACHINE_SIZE = 'f1-micro';
+
+  private readonly apiClient: gcp_api.RestApiClient;
 
+  constructor(private refreshToken: string) {
+    const hardcodedAccessToken = '';
+    // TODO: We need to convert the refreshToken to an access token somewhere.
+    this.apiClient = new gcp_api.RestApiClient(hardcodedAccessToken);
+  }
+
+  /** @see {@link Account#getName}. */
   async getName(): Promise<string> {
-    return 'placeholder';
+    const userInfo = await this.apiClient.getUserInfo();
+    return userInfo.email ?? 'unknown';
```

Comment:
```suggestion
    return userInfo?.email;
```

Let the caller decide what to do in case of missing info.